prototype/FFT_image_processing/serial_port.c

```diff
@@ -0,0 +1,316 @@
+  struct termios tty;
+
+  if (tcgetattr(serialPort, &tty) != 0) {
+    printf("Error %i from tcgetattr: %s\n", errno, strerror(errno));
+    return 1;
+  }
+
+  // set up tty
+  // Control modes
+  tty.c_cflag &= ~(PARENB | PARODD); // Disable parity
+  tty.c_cflag &= ~CSTOPB;            // Only one stop bit
+  tty.c_cflag &= ~CSIZE;             // Clear current byte size
+  tty.c_cflag |= CS8;                // 8 bits per byte
+  tty.c_cflag &= ~CRTSCTS;           // Disable RTS/CTS hardware flow control
+  tty.c_cflag |= (CLOCAL | CREAD);   // Ignore modem controls, enable reading
+
+  // Local modes
+  tty.c_lflag &= ~ICANON; // Disable canonical mode
+  tty.c_lflag &= ~ISIG;   // Disable interpretation of INTR, QUIT and SUSP
+  tty.c_lflag &= ~ECHO;   // Disable echo
+  tty.c_lflag &= ~ECHOE;  // Disable erasure
+  tty.c_lflag &= ~ECHONL; // Disable newline echo
+
+  // Input modes
+  tty.c_iflag &= ~(IXON | IXOFF | IXANY); // Disable software flow control
+  tty.c_iflag &= ~(IGNBRK | BRKINT | ISTRIP | INLCR | IGNCR | ICRNL); // Disable any special handling
+  tty.c_iflag &= ~PARMRK; // Disable parity and framing errors marking
+
+  // Output modes
+  // Prevent special interpretation of output bytes (e.g. newline chars)
+  tty.c_oflag &= ~OPOST;
+  // Prevent conversion of newline to carriage return/line feed
+  tty.c_oflag &= ~ONLCR;
+
+  // VMIN and VTIME have no effect in canonical mode
+  tty.c_cc[VMIN] = 1;  // Wait until one byte is received
+  tty.c_cc[VTIME] = 0; // Wait for up to 0s (Blocking at 0)
+
+  cfsetospeed(&tty, baudRate); // Output baud rate
+  cfsetispeed(&tty, baudRate); // Input baud rate
+  
+  // Save tty settings
+  // Make the change immediatly
+  if (tcsetattr(serialPort, TCSANOW, &tty) != 0) {
+    printf("Error %i from tcsetattr: %s\n", errno, strerror(errno));
+    return 1;
+  }
+
+  return 0;
+}
+
+/**
+ * @brief Read an array of length arraySize, where each element is of size
+ * sizeOfElement. This function will block untill all elements are read.
+ * @param arraySize The size of the array to read.
+ * @param outArray The array in which to store the result of the read data.
+ * @param sizeOfElement The size of each element in the array.
+ * @return Read return code. 0 if everything went correctly.
+ */
+int readArray(const unsigned arraySize, 
+              void* outArray,
+              const unsigned sizeOfElement) {
+  int numBytesWritten = readElement(outArray, arraySize * sizeOfElement);
+  return numBytesWritten;
+}
+
+/**
+ * @brief Write an array of length arraySize, where each element is of size
+ * sizeOfElement. This function will block untill all elements are written.
+ * @param arraySize The size of the array to read.
+ * @param array The array containing the data to be written.
+ * @param sizeOfElement The size of each element in the array.
+ * @return The number of bytes written. -1 if there was an error.
+ */
+int writeArray(const unsigned arraySize, 
```

Comment:
I feel like this method has no real purpose other than adding an overhead to the final code. A lot of method already manipulate byes (for example memcpy or malloc). So why don't we simply have 1 method like `writeSerialData(void* data, size_t length);` that does the work of both writeArray and writeElement?